src/routes/console/project-[project]/messaging/providers/settingsFormList.svelte

```diff
@@ -40,76 +40,81 @@
         | FCMProviderParams
         | APNSProviderParams
     >;
+
+    function getPopoverProps(input: ProviderInput) {
+        if (!input.popover) {
+            return {};
+        }
+        return {
+            lines: input.popover,
+            image: input.popoverImage
+        };
+    }
 </script>
 
 <FormList>
     {#each inputs as input}
-        {@const popoverImage = input.popoverImage}
+        {@const popover = input.popover ? PopoverContent : null}
+        {@const popoverProps = getPopoverProps(input)}
         {#if input.type === 'text'}
             <InputText
                 id={input.name}
                 label={input.label}
                 placeholder={input.placeholder}
                 required={!input.optional}
+                {popover}
+                {popoverProps}
```

Comment:
having to pass the popover like this still feels messy 😕

Reply:
for sure, but at least its explicit.

Having a `popover` slot is not in any relation with the `isPopoverDefined` prop and it will lead to potential errors 👍🏻